### PR TITLE
Peatland rewetting automatically considered during SEALS downscaling

### PR DESCRIPTION
**Pull request by pvjeetze**
## :bird: Description of this PR :bird:

- Reworked `extra/runSEALSallocation.R` so that peatland rewetting is automatically considered during a SEALS allocation run.
- additional data >4.60 includes a SEALS model coefficients file that is modified depending on the scenario settings.

## :wrench: Checklist for PR creator :wrench:

- [x] Label pull request [from the label list](https://github.com/magpiemodel/magpie/labels).
  - **Low risk**: Simple bugfixes (missing files, updated documentation, typos) or changes  in start or output scripts
  - **Medium risk**: Uncritical changes in the model core (e.g. moderate modifications in non-default realizations)
  - **High risk**: Critical changes in model core or default settings (e.g. changing a model default or adjusting a core mechanic in the model)

- [x] Self-review own code
  - No hard coded numbers and cluster/country/region names.
  - The new code doesn't contain declared but unused parameters or variables.
  - [`magpie4`](https://github.com/pik-piam/magpie4) R library has been updated accordingly and backwards compatible where necessary.
  - `scenario_config.csv` has been updated accordingly (important if `default.cfg` has been updated)

- [x] Document changes 
  - Add changes to `CHANGELOG.md`
  - Where relevant, put In-code documentation comments
  - Properly address updates in interfaces in the module documentations
  - run [`goxygen::goxygen()`](https://github.com/pik-piam/goxygen) and verify the modified code is properly documented

- [ ] Perform test runs
  - **Low risk**: 
    - Run a compilation check via `Rscript start.R --> "compilation check"`
  - **Medium risk**: 
    - Run test runs via `Rscript start.R --> "test runs"`
    - Check logs for errors/warnings
  - **High risk**:
    - Run test runs via `Rscript start.R --> "test runs"`
    - Check logs for errors/warnings
    - Default run from the PR target branch for comparison
    - Provide relevant comparison plots (land-use, emissions, food prices, land-use intensity,...)

### :chart_with_downwards_trend: Performance changes :chart_with_upwards_trend:
  
  - Current develop branch default : ** mins
  - This PR's default :  ** mins

## :rotating_light: Checklist for reviewer :rotating_light:

- PR is labeled correctly
- Code changes look reasonable
  - No hard coded numbers and cluster/country/region names.
  - No unnecessary increase in module interfaces
  - model behavior/performance is satisfactory.
- Changes are properly documented
  - `CHANGELOG` is updated correctly
  - Updates in interfaces have been properly addressed in the module documentations
  - In-code documentation looks appropriate
- [ ] content review done (at least 1)
- [ ] RSE review done (at least 1)

### CODE REVIEW

File: scripts/output/extra/runSEALSallocation.R [hunk 1]
```diff
@@ -124,11 +127,94 @@ Sys.chmod(iniLock, mode = "0664")
...
   )
+  sealsCoeff <- suppressWarnings(sealsCoeff[min(which(file.exists(sealsCoeff)))])
+  if (!is.na(sealsCoeff)) {
+    sealsCoeff <- read.csv(sealsCoeff)
+    consvRow <- which(sealsCoeff[, "spatial_regressor_name"] == "land_conservation")
+    sealsCoeff[consvRow, "data_location"] <- sub(
+      "WDPA", consv, sealsCoeff[consvRow, "data_location"]
+    )
+
+    peatlandPrice <- readGDX(file.path(dir, "fulldata.gdx"), "im_pollutant_prices")
```
Comment (flohump) [inline scripts/output/extra/runSEALSallocation.R:167]: I suggest to use a different criterion for activating peatlands in SEALS.
We now also have to option for external peatland rewetting scenarios. 
Therefore, I suggest to use rewetted peatland area as criterion, as a share of total peatland area.
`dimSums(PeatlandArea(gdx)[,,"rewet"],dim=1)/dimSums(PeatlandArea(gdx),dim=c(1,3))>0.01`

**Review by flohump — CHANGES_REQUESTED** [review]
Thanks for including peatlands in SEALS!
I have one request for adjusting the criterion when peatlands are activatd in SEALS.

File: scripts/output/extra/runSEALSallocation.R (hunk 1 above)
Comment (pvjeetze) [reply scripts/output/extra/runSEALSallocation.R]: Thanks for spotting this and the suggestion! I've changed the code based on your suggestion.

File: scripts/output/extra/runSEALSallocation.R [hunk 2]
```diff
@@ -124,11 +127,94 @@ Sys.chmod(iniLock, mode = "0664")
...
+  ### Modify SEALS model coefficients based on scenario settings
+
+  message("Updating SEALS model coefficients based on scenario settings")
+
+  sealsCoeff <- c(
+    "input/seals_global_coefficients.csv",
+    "../input/seals_global_coefficients.csv",
+    "../../input/seals_global_coefficients.csv"
   )
+  sealsCoeff <- suppressWarnings(sealsCoeff[min(which(file.exists(sealsCoeff)))])
```
[the remark this answers is not in the document]
Comment (pvjeetze) [reply scripts/output/extra/runSEALSallocation.R:160]: This approach is pretty common in post-processing functions mostly in the `magpie4` library, from which I have transferred part of the code. This makes the code more stable and less susceptible depending on which relative directory the code may be run.

Would you have an alternative suggestion?

**Review by flohump — APPROVED** [review]
Looks good. Thanks!

File: scripts/output/extra/runSEALSallocation.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (pascal-sauer) [reply scripts/output/extra/runSEALSallocation.R:160]: Maybe a bit nicer this way:
```suggestion
  sealsCoeff <- Find(file.exists, sealsCoeff)
```
This returns NULL if no file exists, so would need to change line 160 to `!is.null`

File: scripts/output/extra/runSEALSallocation.R [hunk 3]
```diff
@@ -124,11 +127,94 @@ Sys.chmod(iniLock, mode = "0664")
...
+
+  ### Modify SEALS model coefficients based on scenario settings
+
+  message("Updating SEALS model coefficients based on scenario settings")
+
+  sealsCoeff <- c(
+    "input/seals_global_coefficients.csv",
+    "../input/seals_global_coefficients.csv",
+    "../../input/seals_global_coefficients.csv"
   )
```
Comment (pascal-sauer) [inline scripts/output/extra/runSEALSallocation.R:159]:
```suggestion
  sealsCoeff <- paste0(c("./", "../", "../../"), "input/seals_global_coefficients.csv")
```

File: scripts/output/extra/runSEALSallocation.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (pascal-sauer) [reply scripts/output/extra/runSEALSallocation.R:160]: I'd suggest the same for seals config below as well

File: scripts/output/extra/runSEALSallocation.R (hunk 3 above)
Comment (pvjeetze) [reply scripts/output/extra/runSEALSallocation.R]: Thanks Pascal for these simplifications! I've reworked the parts of the code. Please double check.

**Review by pascal-sauer — APPROVED** [review]

**Pull request merged**